transform-plugins/src/main/java/io/cdap/plugin/UnionSplitter.java

```diff
@@ -87,17 +87,30 @@ public void transform(StructuredRecord record, MultiOutputEmitter<StructuredReco
     }
 
     Object val = record.get(conf.unionField);
+    Schema.LogicalType logicalType = fieldSchema.getNonNullable().getLogicalType();
     Schema valSchema;
     if (val == null) {
       valSchema = Schema.of(Schema.Type.NULL);
     } else if (val instanceof Boolean) {
       valSchema = Schema.of(Schema.Type.BOOLEAN);
     } else if (val instanceof ByteBuffer || val instanceof byte[] || val instanceof Byte[]) {
-      valSchema = Schema.of(Schema.Type.BYTES);
+      if(logicalType!=null){
```

Comment:
Please fix al the spacing. If statement should be spaced like this:

```
if (condition) {
  statement;
}
```

Reply:
Fixed